zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateParallelGatewayTest.java

```diff
@@ -0,0 +1,108 @@
+  @Rule public final TestWatcher watcher = new RecordingExporterTestWatcher();
+  @Rule public final BrokerClassRuleHelper helper = new BrokerClassRuleHelper();
+
+  @Test
+  public void shouldMigrateParallelGatewayWithIncident() {
+    final String sourceProcessId = helper.getBpmnProcessId();
+    final String targetProcessId = helper.getBpmnProcessId() + "_target";
+
+    final String nonExistingElJobTypeExp = "=source_execution_listener_job_type";
+    final String existingElJobTypeExp = "=target_execution_listener_job_type";
+    final String existingElJobTypeVar = "target_execution_listener_job_type";
+    final String existingElJobTypeValue = "elJobType";
+
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(sourceProcessId)
+                    .startEvent()
+                    .parallelGateway("parallel1")
+                    .zeebeStartExecutionListener(nonExistingElJobTypeExp)
+                    .endEvent("end1")
+                    .moveToLastGateway()
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .parallelGateway("parallel2")
+                    .zeebeStartExecutionListener(existingElJobTypeExp)
+                    .endEvent("end2")
+                    .moveToLastGateway()
+                    .endEvent()
+                    .done())
+            .deploy();
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(sourceProcessId)
+            .withVariable(existingElJobTypeVar, existingElJobTypeValue)
+            .create();
+
+    final var incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId("parallel1")
+            .getFirst();
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("parallel1", "parallel2")
+        .migrate();
+
+    RecordingExporter.incidentRecords(IncidentIntent.MIGRATED)
+        .withRecordKey(incident.getKey())
+        .await();
+
+    // then
+    ENGINE.incident().ofInstance(processInstanceKey).withKey(incident.getKey()).resolve();
+    ENGINE.job().ofInstance(processInstanceKey).withType(existingElJobTypeValue).complete();
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementType(BpmnElementType.END_EVENT)
+                .limit(2))
```

Comment:
🔧 There's a way to limit this on all records of the process instance
```suggestion
    assertThat(
            RecordingExporter.records()
                .limitToProcessInstance(processInstanceKey)
                .processInstanceRecords()
                .withIntent(ProcessInstanceIntent.ELEMENT_ACTIVATED)
                .withProcessInstanceKey(processInstanceKey)
                .withElementType(BpmnElementType.END_EVENT))
```